src/main/java/co/cask/PluginJSON.java

```diff
@@ -256,6 +333,14 @@ private void initialize() {
       widgetDirectory = new File(widgetsDirectory);
     }
 
+    if (iconsDirectory != null) {
+      if (!iconsDirectory.contains("/")) {
+        iconDirectory = new File(project.getBasedir() + "/" + iconsDirectory);
```

Comment:
this will break on windows. You should never manually concatenate '/' for paths. do `new File(baseDirectory, iconsDirectory)`